### PR TITLE
Provide dumps of GCC macros and include paths

### PR DESCRIPTION
**Pull request by martyone**

### CODE REVIEW

File: src/plugins/mer/mertarget.cpp [hunk 1]
```diff
@@ -148,7 +162,8 @@ QVariantMap MerTarget::toMap() const
 
 bool MerTarget::isValid() const
 {
-    return m_sdk && !m_name.isEmpty() && !m_qmakeQuery.isEmpty() && !m_gccMachineDump.isEmpty()
+    return !m_name.isEmpty() && !m_qmakeQuery.isEmpty() && !m_gccMachineDump.isEmpty()
```
Comment (vigejolla) [inline src/plugins/mer/mertarget.cpp]: I don't like the fact that you dropped m_sdk check here. Sure, it should never be null, but I would like to make sure, as programmers are known to make mistakes. I would consider even adding an assert in constructor.

File: src/plugins/mer/mertoolchain.cpp [hunk 2]
```diff
@@ -157,18 +157,17 @@ QList<Task> MerToolChain::validateKit(const Kit *kit) const
...
-    if (m_headerPathsOnHost.isEmpty()) {
-        m_headerPathsOnHost.append(HeaderPath(sysRoot.toString() + QLatin1String("/usr/local/include"),
-                                             HeaderPathType::System));
-        m_headerPathsOnHost.append(HeaderPath(sysRoot.toString() + QLatin1String("/usr/include"),
-                                             HeaderPathType::System));
-    }
-    return m_headerPathsOnHost;
-}
-
+//ProjectExplorer::HeaderPaths MerToolChain::builtInHeaderPaths(const QStringList &cxxflags, const FileName &sysRoot) const
```
Comment (vigejolla) [inline src/plugins/mer/mertoolchain.cpp]: Please remove commented code before removing *WIP* :)

File: src/plugins/mer/mertarget.cpp (hunk 1 above)
Comment (martyone) [reply src/plugins/mer/mertarget.cpp]: Me neither - it's a mistake I did while rebasing, thanks :)

File: src/plugins/mer/mertoolchain.cpp (hunk 2 above)
Comment (martyone) [reply src/plugins/mer/mertoolchain.cpp]: Yup, I just need to recollect why I commented it out :)

**Pull request merged**